APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m

```diff
@@ -0,0 +1,477 @@
+        
+        if (![[NSFileManager defaultManager] createDirectoryAtPath:self.folder
+                                       withIntermediateDirectories:YES
+                                                        attributes:@{
+                                                                     NSFileProtectionKey :
+                                                                         self.fileProtectionKey
+                                                                     }
+                                                             error:&folderCreationError])
+        {
+            APCLogError2(folderCreationError);
+        }
+        else
+        {
+            [self resetDataFilesForTracker];
+        }
+    }
+    
+    NSData* dictData = [NSData dataWithContentsOfFile:infoFilePath];
+
+    infoDictionary      = [NSDictionary dictionaryWithJSONString:[[NSString alloc] initWithData:dictData
+                                                                                       encoding:NSUTF8StringEncoding]];
+    self.infoDictionary = infoDictionary;
+}
+
+- (void) checkIfCSVStructureHasChanged
+{
+    NSString* csvFilePath = [self.folder stringByAppendingPathComponent:kCSVFilename];
+    
+    if ([[NSFileManager defaultManager] fileExistsAtPath:csvFilePath])
+    {
+        // read everything from text
+        NSError*    fileContentsError   = nil;
+        NSString*   fileContents        = [NSString stringWithContentsOfFile:csvFilePath
+                                                                    encoding:NSUTF8StringEncoding
+                                                                       error:&fileContentsError];
+        
+        if (fileContentsError)
+        {
+            APCLogError2(fileContentsError);
+        }
+        
+        // first, separate by new line
+        NSArray* dataSeparatedByNewLine = [fileContents componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]];
+        
+        if (dataSeparatedByNewLine.count > 0)
+        {
+            NSString*   expectedColumn          = [dataSeparatedByNewLine objectAtIndex:0];
+            NSArray*    oldColumnStructure      = [expectedColumn componentsSeparatedByString:@","];
+            NSArray*    possibleNewStructure    = [NSArray arrayWithArray:self.columnNames];
+            
+            if (![oldColumnStructure isEqualToArray:possibleNewStructure] && dataSeparatedByNewLine.count == 1)
+            {
+                //If there isn't data then reset the file.
+                [self resetDataFilesForTracker];
+            }
+            else
+            {
+                //If there's data then upload this data.
+                [self flush];
+            }
+        }
+        else
+        {
+            //If the file exists and there are no columns reset the file.
+            [self resetDataFilesForTracker];
+        }
+    }
+}
+
+- (void) checkIfDataNeedsToBeFlushed
```

Comment:
Method name is misleading: the data is flushed if needed and not just checking if the need is there.